### ISSUE TITLE
`max-len`: passing a RegExp to `ignorePattern` doesn't work

### ISSUE THREAD

**Issue by simon-abbott** [eslint-stylistic/eslint-stylistic#868]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Passing a `RegExp` to the `ignorePattern` option of `@stylistic/max-len` throws an error when running ESLint, and doesn't actually result in anything being ignored.

### Reproduction

https://stackblitz.com/edit/github-3qtfaowc?file=eslint.config.js

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
Hi! Thanks for the issue.

I'm afraid that `string` should be used instead of `RegExp`, `ESLint` uses `ajv` to check the configurations, and `RegExp` is not supported.
This rule uses `new RegExp` to transform the `ignorePattern`.

This works:
``` js
'@stylistic/max-len': ['error', { comments: 80, ignorePattern: "eslint-disable" }]
```

**Comment by simon-abbott** [comment]
Yeah, I figured out that strings work, I just assumed from the way the documentation was written that RegExp should also work when defined in a JS config file. Maybe it would be a good idea to update the docs to make that clear? Especially now that ESLint is moving fully to the JS-based flat config.

**Comment by 9romise** [comment]
Agreed, I think it makes sense to point this out in the docs, would you be interested in creating a PR to improve it?

**Comment by simon-abbott** [comment]
I'm willing to, but I don't know when I will have time, so if someone else gets to it first be my guest.
